testing/jest/scheduler/appointment-tooltip/tooltip-item-layout.tests.tsx

```diff
@@ -0,0 +1,326 @@
+import { h } from 'preact';
+import { shallow } from 'enzyme';
+import { Fragment } from 'devextreme-generator/component_declaration/common';
+import { dxSchedulerAppointment } from '../../../../js/ui/scheduler';
+import TooltipItemLayout, {
+  viewFunction as TooltipItemLayoutView,
+  TooltipItemLayoutProps,
+} from '../../../../js/renovation/scheduler/appointment-tooltip/tooltip-item-layout';
+import DeleteButton from '../../../../js/renovation/button';
+import Marker from '../../../../js/renovation/scheduler/appointment-tooltip/marker';
+import TooltipItemContent from '../../../../js/renovation/scheduler/appointment-tooltip/tooltip-item-content';
+
+jest.mock('../../../../js/renovation/button', () => () => null);
+jest.mock('../../../../js/renovation/scheduler/appointment-tooltip/marker', () => () => null);
+jest.mock('../../../../js/renovation/scheduler/appointment-tooltip/tooltip-item-content', () => () => null);
+
+describe('TooltipItemLayout', () => {
+  describe('View', () => {
+    const defaultProps: TooltipItemLayoutProps = {
+      onDelete: jest.fn(),
+      onHide: jest.fn(),
+      getTextAndFormatDate: jest.fn(() => ({
+        text: 'Appointment text',
+        formatDate: 'Formatted date',
+      })),
+      showDeleteButton: true,
+      item: {
+        data: { text: 'data' },
+        currentData: { text: 'currentData' },
+      },
+      index: 0,
+      singleAppointmentData: {
+        text: 'singleAppointmentData',
+      },
+    };
+
+    it('should combine `className` with predefined classes', () => {
+      const tree = shallow(<TooltipItemLayoutView
+        props={{ ...defaultProps, className: 'custom-class' }}
+      />).childAt(0);
+
+      expect(tree.hasClass('dx-tooltip-appointment-item'))
+        .toBe(true);
+      expect(tree.hasClass('custom-class'))
+        .toBe(true);
+    });
+
+    it('should spread restAttributes', () => {
+      const tree = shallow(<TooltipItemLayoutView
+        restAttributes={{ customAttribute: 'customAttribute' }}
+        props={defaultProps}
+      />);
+
+      expect(tree.childAt(0).prop('customAttribute'))
+        .toBe('customAttribute');
+    });
```

Comment:
All test with `restAttributes` It can be like that
```js
const tree = shallow(<ErrorMessage />);
expect(tree.prop('restAttributes')).toBe('restAttributes');
```